src/Relation/Binary/HeterogeneousEquality.agda

```diff
@@ -240,16 +240,16 @@ module ≅-Reasoning where
   start : ∀ {x : A} {y : B} → x IsRelatedTo y → x ≅ y
   start (relTo x≅y) = x≅y
 
-  ≡-go : ∀ {A : Set a} → Trans {A = A} {C = A} _≡_ _IsRelatedTo_ _IsRelatedTo_
+  ≡-go : ∀ {A : Set ℓ} {B : Set ℓ} → Trans {A = A} {C = B} _≡_ _IsRelatedTo_ _IsRelatedTo_
```

Comment:
This, I think, will still not have as much level polymorphism as you intend with this PR: level `ℓ` will be generalised, as per the `variable` declarations governing the module, but `A` and `B`, being explicitly mentioned in the type here, will each be bound *at the same level*.
```suggestion
  ≡-go : ∀ {A : Set a} {B : Set b} → Trans {A = A} {C = B} _≡_ _IsRelatedTo_ _IsRelatedTo_
```

Reply:
Right, I forgot that HEq also supports different levels. I just modified `_IsRelatedTo_`, `≡-go`, and the anonymous module accordingly.